projects/addon-editor/extensions/iframe-editor/iframe-editor.component.ts

```diff
@@ -0,0 +1,108 @@
+import {DOCUMENT} from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    HostBinding,
+    Inject,
+    Self,
+    ViewChild,
+} from '@angular/core';
+import {DomSanitizer, SafeResourceUrl} from '@angular/platform-browser';
+import {TuiNodeViewNgComponent} from '@taiga-ui/addon-editor/extensions/tiptap-node-view';
+import {TuiDestroyService, tuiIsNumber, tuiTypedFromEvent} from '@taiga-ui/cdk';
+import {merge} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+
+import {
+    TUI_IFRAME_EDITOR_OPTIONS,
+    TuiIframeEditorBoxOptions,
+    TuiIframeEditorOptions,
+} from './iframe-options';
+
+@Component({
+    selector: 'tui-iframe-editor',
+    templateUrl: './image-iframe.component.html',
+    styleUrls: ['./image-iframe.component.less'],
+    providers: [TuiDestroyService],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiIframeEditorComponent extends TuiNodeViewNgComponent {
+    @ViewChild('iframe')
+    private readonly iframe?: ElementRef<HTMLIFrameElement>;
+
+    private initialized = false;
+    private _height = 0;
+    private _width = 0;
+
+    @HostBinding('style.width')
+    hostWidth: string | null = null;
+
+    get attrs(): TuiIframeEditorOptions {
+        return (this.node?.attrs as any) || {src: ''};
+    }
+
+    get src(): SafeResourceUrl {
+        return this.sanitizer.bypassSecurityTrustResourceUrl(this.attrs.src ?? '');
+    }
+
+    get width(): number | string | null {
+        return this._width || this.attrs.width || null;
+    }
+
+    get height(): number | string | null {
+        return this._height || this.attrs.height || null;
+    }
+
+    constructor(
+        @Inject(TUI_IFRAME_EDITOR_OPTIONS) readonly options: TuiIframeEditorBoxOptions,
+        @Inject(DOCUMENT) readonly documentRef: Document,
+        @Inject(DomSanitizer) private readonly sanitizer: DomSanitizer,
+        @Self()
+        @Inject(TuiDestroyService)
+        readonly destroy$: TuiDestroyService,
+    ) {
+        super();
+
+        merge(
+            tuiTypedFromEvent(this.documentRef, 'touchend'),
+            tuiTypedFromEvent(this.documentRef, 'mouseup'),
+        )
+            .pipe(takeUntil(destroy$))
+            .subscribe(() =>
+                this.updateAttributes({width: this.width, height: this.height}),
```

Comment:
There are many code duplications with [ImageEditor](https://github.com/Tinkoff/taiga-ui/blob/main/projects/addon-editor/extensions/image-editor/image-editor.component.ts).

Could we solve it somehow ? Maybe some abstract class to extend (or something else)?

Reply:
fixed